pkg/quota/image/imagestream_evaluator.go

```diff
@@ -84,11 +121,59 @@ func (c *imageStreamUsageComputer) Usage(object runtime.Object) kapi.ResourceLis
 		return kapi.ResourceList{}
 	}
 
+	return map[kapi.ResourceName]resource.Quantity{
+		imageapi.ResourceProjectImagesSize: *c.GetImageStreamSize(is),
+		imageapi.ResourceImageStreamSize:   *resource.NewQuantity(0, resource.BinarySI),
+		imageapi.ResourceImageSize:         *resource.NewQuantity(0, resource.BinarySI),
+	}
+}
+
+// makeImageStreamUsageComputerFactory returns an object used during computation of image quota across all
+// repositories in a namespace.
+func makeImageStreamUsageComputerForAdmissionFactory(osClient osclient.Interface) quotautil.UsageComputerFactory {
+	return func() quotautil.UsageComputer {
+		return &imageStreamUsageComputerForAdmission{
+			GenericImageStreamUsageComputer: *NewGenericImageStreamUsageComputer(osClient, true, false),
+		}
+	}
+}
+
+// imageStreamUsageComputerForAdmission is a context object for use in SharedContextEvaluator
+type imageStreamUsageComputerForAdmission struct {
+	GenericImageStreamUsageComputer
+}
+
+// Usage returns a usage for an image stream.
+func (c *imageStreamUsageComputerForAdmission) Usage(object runtime.Object) kapi.ResourceList {
+	newIS, ok := object.(*imageapi.ImageStream)
+	if !ok {
+		return kapi.ResourceList{}
+	}
+
+	oldIS, err := c.getImageStream(newIS.Namespace, newIS.Name)
+	if err != nil {
+		if !kerrors.IsNotFound(err) && newIS.Namespace != "" {
```

Comment:
This block of code doesn't appear in the final commit. I'll squash this with the following one.